Provide systemd unit file in `.deb` and `.rpm` packages (#4478)
FerretDB/FerretDB: website/docs/quickstart-guide/systemd.md

Reviewer: @Fashander PTAL

---
sidebar_position: 4
---

# Systemd Unit

With both DEB and RPM package we ship the systemd unit, to start FerretDB automatically.

> If the FerretDB is not installed yet, please refer to the one of the matching pages.

Reviewer: Links?

> If the FerretDB is not installed yet, please refer to the one of the matching pages.

Reviewer 2: ```suggestion
If FerretDB is not installed yet, please refer to the [`.deb`](https://docs.ferretdb.io/quickstart-guide/deb/) or [`.rpm`](https://docs.ferretdb.io/quickstart-guide/rpm/) installation pages.
```

The unit file provides some simple environment variables as an example. 
They should be overwritten with the proper configuration, with the [correct flags](../configuration/flags.md).

```
[Service]
ExecStart=/usr/bin/ferretdb
Restart=on-failure

# Configure the FerretDB service with `systemctl edit ferretdb`.
# For more configuration options check https://docs.ferretdb.io/configuration/flags/

Environment="FERRETDB_POSTGRESQL_URL=postgres://username:password@127.0.0.1:5432/ferretdb"
Environment="FERRETDB_LISTEN_ADDR=:27017"
Environment="FERRETDB_DEBUG_ADDR=:8088"
```

> You can modify them by using systemctl edit ferretdb command. This'll open a text editor that'll allow you to

Reviewer: ```suggestion
You can modify them by using `systemctl edit ferretdb` command.
This'll open a text editor that'll allow you to
```

override the systemd options of choice.
For example, if we want to use SQLite backend instead of PostgreSQL, we could write something like:

> ```

Reviewer 2: what's the file format for this? "```text"?

Reviewer: But it should be added to `additionalLanguages` in Docusaurus config

---
sidebar_position: 4
---

# Systemd Unit

:::note
This feature is still experimental.
If you encounter any problem, please [join our community](/#community) to report it.
:::

With both DEB and RPM package we ship the systemd unit, to start FerretDB automatically.
If FerretDB is not installed yet, please refer to the [`.deb`](https://docs.ferretdb.io/quickstart-guide/deb/) or [`.rpm`](https://docs.ferretdb.io/quickstart-guide/rpm/) installation pages.

The unit file provides some basic environment variables as an example.

> They should be overwritten with the proper configuration, with the [correct flags](../configuration/flags.md).

Reviewer: Not flags, though

```systemd
[Service]
ExecStart=/usr/bin/ferretdb
Restart=on-failure

# Configure the FerretDB service with `systemctl edit ferretdb`.
# For more configuration options check https://docs.ferretdb.io/configuration/flags/

Environment="FERRETDB_POSTGRESQL_URL=postgres://username:password@127.0.0.1:5432/ferretdb"
```

You can modify them by using `systemctl edit ferretdb` command.
This'll open a text editor that'll allow you to override the systemd options of choice.
For example, if we want to use SQLite backend instead of PostgreSQL, we could write something like:

```systemd
### Editing /etc/systemd/system/ferretdb.service.d/override.conf
### Anything between here and the comment below will become the new contents of the file

[Service]
Environment="FERRETDB_SQLITE_URL=file:/var/lib/ferretdb/data/"
Environment="FERRETDB_HANDLER=sqlite"

> Environment="FERRETDB_POSTGRESQL_URL="

Reviewer: Why do we need that line?

Author: I've mostly wanted to show how the value can be unset but yeah, right now its not that needed
